xendit-android/src/main/java/com/xendit/TokenBroadcastReceiver.java

```diff
@@ -32,7 +32,11 @@ public void onReceive(Context context, Intent intent) {
             if (authentication.getStatus().equals("VERIFIED")) {
                 tokenCallback.onSuccess(new Token(authentication));
             } else {
-                tokenCallback.onError(new XenditError(context.getString(R.string.create_token_error_validation), authentication));
+                if (message.contains("AUTHENTICATION_FAILED")) {
```

Comment:
Why don't we handle this based on status code? If it's 200, we return the authentication object. if it's anything else and it's a json object, we return the error object with code and original message?